doc/api/single-executable-applications.md

```diff
@@ -0,0 +1,44 @@
+# Single executable applications
+
+<!--introduced_in=REPLACEME-->
+
+> Stability: 1 - Experimental
+
+> This feature is currently being designed and will still change.
+
+<!-- source_link=lib/internal/main/single_executable_application.js -->
+
+Node.js supports the creation of single executable applications by allowing the
+injection of a JavaScript file into the binary. During the startup phase, the
+program checks if a resource (on [PE][]) / section (on [Mach-O][] or [ELF][])
+named `NODE_JS_CODE` exists. If it is found, it executes its contents, otherwise
+it operates like plain Node.js.
+
+This feature allows the distribution of a Node.js application conveniently to a
+system that does not have Node.js installed.
+
+A bundled JavaScript file can be turned into a single executable application
```

Comment:
We should probably also add a comment that it can also be done with any other tool which can inject sections.

Reply:
How can we use any other tools? Can you add them to the instructions?

Reply:
`objcopy` can be used to accomplish this. I've created an issue in postject to discuss if it makes sense to use it internally: https://github.com/nodejs/postject/issues/63.

However, after https://github.com/nodejs/postject/pull/59 lands, it would also add another step - flipping the bits in the fuse. This can't be done purely by a section injection tool.